nisystemlink/clients/spec/utilities/_dataframe_utilities.py

```diff
@@ -0,0 +1,319 @@
+from typing import Callable, Dict, List, Optional, Union
+
+import pandas as pd
+from nisystemlink.clients.spec._spec_client import SpecClient
+from nisystemlink.clients.spec.models._condition import (
+    Condition,
+    NumericConditionValue,
+    StringConditionValue,
+)
+from nisystemlink.clients.spec.models._query_specs import (
+    SpecificationProjection,
+)
+from nisystemlink.clients.spec.models._specification import Specification
+from nisystemlink.clients.spec.utilities._client_utilities import __batch_query_specs
+from nisystemlink.clients.spec.utilities._constants import (
+    CONDITION_COLUMN_HEADER_PREFIX,
+    KEYWORDS_COLUMN_HEADER,
+    PROPERTY_COLUMN_HEADER_PREFIX,
+)
+
+
+def __serialize_conditions(conditions: List[Condition]) -> Dict[str, str]:
+    """Serialize conditions into desired format.
+
+    Args:
+        conditions: List of all conditions in a spec.
+
+    Returns:
+        Conditions as a dictionary.
+    """
+    return {
+        __generate_condition_column_header(condition): ", ".join(
+            __get_condition_values(condition)
+        )
+        for condition in conditions
+    }
+
+
+def get_specs_dataframe(
+    client: SpecClient,
+    product_id: str,
```

Comment:
The underlying query API supports multiple products. Is there a reason you'd only want to support one in this function?